docker/Dockerfile.hipcc

```diff
@@ -71,21 +81,20 @@ RUN BOOST_VERSION=1.72.0 && \
         cxxflags=-w \
         install \
         && \
-    rm -rf boost*
+    rm -rf ${SCRATCH_DIR}
 
 # Install Google Benchmark support library
 ENV BENCHMARK_DIR=/opt/benchmark
 RUN SCRATCH_DIR=/scratch && mkdir -p ${SCRATCH_DIR} && cd ${SCRATCH_DIR} && \
-    git clone https://github.com/google/benchmark.git -b v1.5.0 && \
+    git clone https://github.com/google/benchmark.git -b v1.4.1 && \
```

Comment:
What the point of that change

Reply:
All these changes are so that I get
```diff
diff Dockerfile Dockerfile.hipcc 
1c1
< ARG BASE=nvidia/cuda:10.1-devel
---
> ARG BASE=rocm/dev-ubuntu-20.04:3.8
5a6,8
> # Avoid tzdata dialog
> ARG DEBIAN_FRONTEND=noninteractive
> 
10a14
>         kmod \
14,15d17
<         lcov \
<         ccache \
17c19
<         ninja-build \
---
>         ccache \
21a24
>         libopenmpi-dev \
25a29,30
> ENV PATH=/opt/rocm/bin:$PATH
> 
51,86d55
< # Install Clang/LLVM
< ENV LLVM_DIR=/opt/llvm
< RUN LLVM_VERSION=10.0.0 && \
<     LLVM_KEY="86419D8A 345AD05D" && \
<     LLVM_URL=https://github.com/llvm/llvm-project/releases/download/llvmorg-${LLVM_VERSION}/clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-18.04.tar.xz && \
<     LLVM_ARCHIVE=llvm-${LLVM_VERSION}.tar.xz && \
<     SCRATCH_DIR=/scratch && mkdir -p ${SCRATCH_DIR} && cd ${SCRATCH_DIR} && \
<     wget --quiet ${LLVM_URL} --output-document=${LLVM_ARCHIVE} && \
<     wget --quiet ${LLVM_URL}.sig --output-document=${LLVM_ARCHIVE}.sig && \
<     gpg --verify ${LLVM_ARCHIVE}.sig ${LLVM_ARCHIVE} && \
<     mkdir -p ${LLVM_DIR} && \
<     tar -xvf ${LLVM_ARCHIVE} -C ${LLVM_DIR} --strip-components=1 && \
<     echo "${LLVM_DIR}/lib" > /etc/ld.so.conf.d/llvm.conf && ldconfig && \
<     rm -rf ${SCRATCH_DIR}
< ENV PATH=${LLVM_DIR}/bin:$PATH
< 
< # Install OpenMPI
< ARG CUDA_AWARE_MPI
< ENV OPENMPI_DIR=/opt/openmpi
< RUN OPENMPI_VERSION=4.0.3 && \
<     OPENMPI_VERSION_SHORT=$(echo "$OPENMPI_VERSION" | cut -d. -f1,2) && \
<     OPENMPI_SHA1=d958454e32da2c86dd32b7d557cf9a401f0a08d3 && \
<     OPENMPI_URL=https://download.open-mpi.org/release/open-mpi/v${OPENMPI_VERSION_SHORT}/openmpi-${OPENMPI_VERSION}.tar.bz2 && \
<     OPENMPI_ARCHIVE=openmpi-${OPENMPI_VERSION}.tar.bz2 && \
<     CUDA_OPTIONS=${CUDA_AWARE_MPI:+--with-cuda} && \
<     SCRATCH_DIR=/scratch && mkdir -p ${SCRATCH_DIR} && cd ${SCRATCH_DIR} && \
<     wget --quiet ${OPENMPI_URL} --output-document=${OPENMPI_ARCHIVE} && \
<     echo "${OPENMPI_SHA1} ${OPENMPI_ARCHIVE}" | sha1sum -c && \
<     mkdir -p openmpi && \
<     tar -xf ${OPENMPI_ARCHIVE} -C openmpi --strip-components=1 && \
<     mkdir -p build && cd build && \
<     ../openmpi/configure --prefix=${OPENMPI_DIR} ${CUDA_OPTIONS} CFLAGS=-w && \
<     make -j${NPROCS} install && \
<     rm -rf ${SCRATCH_DIR}
< ENV PATH=${OPENMPI_DIR}/bin:$PATH
< 
89c58
< RUN BOOST_VERSION=1.67.0 && \
---
> RUN BOOST_VERSION=1.72.0 && \
108d76
<     echo "using mpi ;" >> project-config.jam && \
128,130d95
< # Workaround for Kokkos to find libcudart
< ENV LD_LIBRARY_PATH=/usr/local/cuda/targets/x86_64-linux/lib:${LD_LIBRARY_PATH}
< 
132,133c97,98
< ARG KOKKOS_VERSION=3.1.00
< ARG KOKKOS_OPTIONS="-DKokkos_ENABLE_SERIAL=ON -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ENABLE_CUDA_LAMBDA=ON"
---
> ARG KOKKOS_VERSION=3.2.00
> ARG KOKKOS_OPTIONS="-DKokkos_ENABLE_HIP=ON -DKokkos_ARCH_VEGA906=ON"
143c108
<     cmake -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=${KOKKOS_DIR} -D CMAKE_CXX_COMPILER=/scratch/kokkos/bin/nvcc_wrapper ${KOKKOS_OPTIONS} .. && \
---
>     cmake -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=${KOKKOS_DIR} -D CMAKE_CXX_COMPILER=hipcc ${KOKKOS_OPTIONS} .. && \
```
and the differences are small. We still have a different boost version and we don't use MPI but other than that's it's pretty much minimal.

Reply:
Personally, when we are talking about minimizing differences between two docker files, this only concerns the meta aspects of it such as downloading repos, unpacking, building, keys, etc. It should not concern versioning, imho.

Reply:
Maybe I should not have chosen this comment to respond but one of the others.
Apparently, it doesn't really matter so the question is if you want me to add a commit to change it back or not.

Reply:
I would suggest changing it back to 1.5.0 if only to increase the testing coverage.

Reply:
If no one requests it, I won't change. @dalg24 Was that a request to change this one or just a question?

Reply:
Please change.

Reply:
Here you go.